dmscripts/mark_definite_framework_results.py

```diff
@@ -11,37 +11,40 @@ def _passes_validation(candidate, schema, logger, schema_name="schema", tablevel
     except jsonschema.ValidationError as e:
         # extracting multiple errors from a single validation is possible but a bit more involved than i'm willing
         # to go right now
-        logger.log(loglevel, "%sFailed %s @ %s: %s", "\t"*tablevel, schema_name, "/".join(e.absolute_path), e.message)
+        logger.log(loglevel, "%sFailed %s @ %s: %s", "\t" * tablevel, schema_name, "/".join(e.absolute_path), e.message)
         return False
     else:
         return True
 
 
 def _assess_draft_services(
-        client,
-        updated_by,
-        framework_slug,
-        supplier_id,
-        service_schema=None,
-        dry_run=True,
-        reassess_failed_draft_services=False,
-        logger=logging.getLogger("script"),
-        ):
+    client,
+    updated_by,
+    framework_slug,
+    supplier_id,
+    service_schema=None,
+    dry_run=True,
+    reassess_failed_draft_services=False,
+    logger=logging.getLogger("script"),
+):
     counter = Counter()
     for draft_service in client.find_draft_services_iter(supplier_id, framework=framework_slug):
         if draft_service["status"] in ("submitted", "failed",):
             logger.debug("\tAssessing draft service %s", draft_service["id"])
             if draft_service["status"] == "failed" and not reassess_failed_draft_services:
                 logger.debug("\t\tSkipping - already marked failed")
                 counter["skipped"] += 1
-            elif service_schema is None or _passes_validation(
+            elif (
+                service_schema is None
+                or _passes_validation(
                     draft_service,
                     service_schema,
                     logger,
                     schema_name="service_schema",
                     tablevel=2,
-                    loglevel=logging.DEBUG,
-                    ):
+                    loglevel=logging.DEBUG
+                )
+            ):
```

Comment:
Doesn't flake8 complain about these outer parenthesis being redundant? They aren't enforcing anything.

Reply:
Implied line continuation. Otherwise we'd need a continuation character `\` which aren't preferred:
https://www.python.org/dev/peps/pep-0008/#maximum-line-length

Reply:
I'd have written it like:
```
    for draft_service in client.find_draft_services_iter(supplier_id, framework=framework_slug):
        if draft_service["status"] in ("submitted", "failed",):
            logger.debug("\tAssessing draft service %s", draft_service["id"])
            if draft_service["status"] == "failed" and not reassess_failed_draft_services:
                logger.debug("\t\tSkipping - already marked failed")
                counter["skipped"] += 1
                continue

            # Initial validation check
            passes_validation = service_schema is None
            if not passes_validation:
                # Second chance validation check
                passes_validation = _passes_validation(
                    draft_service,
                    service_schema,
                    logger,
                    schema_name="service_schema",
                    tablevel=2,
                    loglevel=logging.DEBUG
                )
            if passes_validation:
                if not dry_run:
                    # mark this as submitted
                    if draft_service["status"] != "submitted":
                        client.update_draft_service_status(draft_service["id"], "submitted", updated_by)
                    else:
                        logger.debug("\tUnchanged result - not re-setting")
                counter["passed"] += 1
            else:
                if not dry_run:
                    # mark this as failed
                    if draft_service["status"] != "failed":
                        client.update_draft_service_status(draft_service["id"], "failed", updated_by)
                    else:
                        logger.debug("\tUnchanged result - not re-setting")
                counter["failed"] += 1
```